tools/test-npm.ps1

```diff
@@ -0,0 +1,56 @@
+#Port of tools/test-npm.sh to windows
+
+# Handle parameters
+param (
+  [string]$progress = "classic",
+  [string]$logfile
+)
+
+#always change the working directory to the project's root directory
+$dp0 = (Get-Item -Path ".\" -Verbose).FullName
+cd $~dp0\..
+
+# Use rmdir to get around long file path issues
+Cmd /C "rmdir /S /Q test-npm"
+Remove-Item test-npm.tap -ErrorAction SilentlyContinue
+
+#make a copy of deps/npm to run the tests on
+Copy-Item deps\npm test-npm -Recurse
+cd test-npm
+
+# do a rm first just in case deps/npm contained these
+Remove-Item npm-cache -Force -Recurse -ErrorAction SilentlyContinue
+Remove-Item npm-tmp -Force -Recurse -ErrorAction SilentlyContinue
+Remove-Item npm-prefix -Force -Recurse -ErrorAction SilentlyContinue
+
+New-Item -ItemType directory -Path npm-cache
+New-Item -ItemType directory -Path npm-tmp
+New-Item -ItemType directory -Path npm-prefix
+
+#set some npm env variables to point to our new temporary folders
+$pwd = (Get-Item -Path ".\" -Verbose).FullName
+Set-Variable -Name npm_config_cache -value ("$pwd\npm-cache") -Scope Global
+Set-Variable -Name npm_config_prefix -value ("$pwd\npm-prefix") -Scope Global
+Set-Variable -Name npm_config_tmp -value ("$pwd\npm-tmp") -Scope Global
```

Comment:
Shouldn't `npm_config_userconfig` be set here as well? I've had issues in a CI job that ran in a machine with a conflicting `.npmrc` in the default location, and this solved it.